src/main/php/org/bovigo/vfs/vfsStreamWrapper.php

```diff
@@ -385,7 +390,10 @@ private function createFile(string $path, string $mode = null, int $options = nu
             return false;
         }
 
-        return vfsStream::newFile($names['basename'])->at($dir);
+        /** @var vfsStreamFile $file */
+        $file = vfsStream::newFile($names['basename'])->at($dir);
```

Comment:
Simply out of curiosity, what's the benefit of storing this in a variable instead of returning it immediately?

Reply:
It's purely to add the `/** @var vfsStreamFile $file */` doc so phpstan knows what is being returned. 

The `at()` method is type hinted to only return `vfsStreamContent` and the method this is being returned in expects a `vfsStreamFile` (or `false`). Although `vfsStreamFile` is a `vfsStreamContent`, `vfsStreamContent` is not a `vfsStreamFile` so phpstan sees it as an error.

Reply:
Would phpstan respect a `@return static` to allow for the type hinting of `at()`?

Reply:
It doesn't. `at()` has an actual return type in the method signature, so it supersedes the doc block's return type.